tests/Unit/ApparentHorizons/Test_Tags.cpp

```diff
@@ -299,6 +305,62 @@ void test_min_ricci_scalar() {
   CHECK(expected_min == min);
 }
 
+void test_dimensionful_spin_vector_compute_tag() noexcept {
+  const double y11_amplitude = 1.0;
+  const double y11_radius = 2.0;
+  const std::array<double, 3> center = {{0.1, 0.2, 0.3}};
+  // create a strahlkorper
+  const auto strahlkorper =
+      create_strahlkorper_y11(y11_amplitude, y11_radius, center);
+  const size_t ylm_physical_size =
+      strahlkorper.ylm_spherepack().physical_size();
+  const DataVector used_for_size(ylm_physical_size,
+                                 std::numeric_limits<double>::signaling_NaN());
+
+  // Creates a variable named generator that can be used to generate random
+  // values
+  MAKE_GENERATOR(generator);
+  // Creates a uniform distribution, which will be used to generate random
+  // numbers
+  std::uniform_real_distribution<> dist(-1., 1.);
+
+  const double dimensionful_spin_magnitude{5.0};
+  // Create the tensor arguments to spin_vector by having them contain
+  // DataVectors of size == ylm physical size, where values are random
+  const auto area_element = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), dist, used_for_size);
+  const auto radius = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), dist, used_for_size);
+  const auto r_hat =
+      make_with_random_values<tnsr::i<DataVector, 3, Frame::Inertial>>(
+          make_not_null(&generator), dist, used_for_size);
+  const auto ricci_scalar = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), dist, used_for_size);
+  const auto spin_function = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), dist, used_for_size);
+  const auto box = db::create<
+      db::AddSimpleTags<StrahlkorperGr::Tags::DimensionfulSpinMagnitude,
+                        StrahlkorperGr::Tags::AreaElement<Frame::Inertial>,
+                        StrahlkorperTags::Radius<Frame::Inertial>,
+                        StrahlkorperTags::Rhat<Frame::Inertial>,
+                        StrahlkorperTags::RicciScalar,
+                        StrahlkorperGr::Tags::SpinFunction,
+                        StrahlkorperTags::Strahlkorper<Frame::Inertial>>,
+      db::AddComputeTags<StrahlkorperGr::Tags::DimensionfulSpinVectorCompute<
+          Frame::Inertial>>>(dimensionful_spin_magnitude, area_element, radius,
+                             r_hat, ricci_scalar, spin_function, strahlkorper);
+  // LHS of the == in the CHECK is retrieving the computed dimensionful spin
+  // vector from your DimensionfulSpinVectorCompute tag and RHS of ==
+  // should be same logic as DimensionfulSpinVectorCompute::function
+  const bool check =
+      db::get<StrahlkorperGr::Tags::DimensionfulSpinVector<Frame::Inertial>>(
+          box) ==
+      StrahlkorperGr::spin_vector<Frame::Inertial>(
+          dimensionful_spin_magnitude, area_element, radius,
+          r_hat, ricci_scalar, spin_function, strahlkorper);
+  CHECK(check);
```

Comment:
[nitpick] why create an extra bool?  It seems simpler and easier to read to just have `CHECK(db::get<StrahlkorperGr...etc);`

Reply:
You are right, that is a cleaner edit, change has been made @markscheel